### DISCUSSION TITLE
Display Update Notice To User

### DISCUSSION THREAD

**Discussion by rootzoll** [raspiblitz/raspiblitz#3946] [Feature / App Ideas]
Tho Nodl has a clever way to get some basic information if a update is available - without the node having to call home.

https://medium.com/@ketominer/how-the-nodl-polls-for-updates-without-us-knowing-and-other-privacy-concerns-ffa7871c8797

For the beginning (v1.1) it would make sense to have just a extra optione in the main menu "Check for Updates" that makes that check and shows some instructions just in case .. maybe even offering to prepare for update by making some extra backup data and turniung the device off. In the future also on the LCD should be some notice about an available update (maybe just turn the color of the version red or something like that).

**Comment by nyxnor** [comment]
If fulmo hosts an onion service, with a page displaying Blitz version, this could be queried to verify with full security and anonimity. Also, a daily script to github with socks5 would also work (need to put latest tag on the release).
```
curl --connect-timeout 10 --socks5-hostname 127.0.0.1:9050 -s https://api.github.com/repos/rootzoll/raspiblitz/releases/latest|grep tag_name|head -1|cut -d '"' -f4
```
The best option if to curl from a onion service, the one I created but can be hosted by anyone:
```
blitzVersion=$(cat /home/admin/_version.info); curlBlitzVersion=$(torsocks curl http://kq3nkqhpcb3qwxy7nc45hvimruchynvvhmyce4g2ohvcq5pqzl3ih3id.onion/nyxnor/blitzVersion.txt); if [ $blitzVersion != $curlBlitzVersion ]; then echo "You are late, update. You are in version '$blitzVersion', Blitz repo is in version '$curlBlitzVersion'."; else echo "Up to date version '$blitzVersion'"; fi
```

https://gitlab.lightning-solutions.eu/nodl/nodl-docs/-/blob/master/docs/guides/upgrade-nodl.md
https://gitlab.lightning-solutions.eu/nodl-private/nodl-admin-private/-/blob/nodl-dojo/bin/nodl-installer.php

They dont do through Tor. Ketominer explained some users with slow internet fail to update with Tor. They have other privacy measure that to not be biased, here is a compilation of the responses till this moment.

**Comment by nyxnor** [comment]
https://github(dot)com/bitcoin/bitcoin/issues/2626 - Change (dot) to (.)
Update checker - Bitcoin core
